lib/improver/blending/spatial_weights.py

```diff
@@ -0,0 +1,275 @@
+                This coordinate corresponds to the dimension along which the
+                normalisation is needed.
+
+        Returns:
+            result (iris.cube.Cube):
+                A cube with the same dimensions as the input cube, but with
+                the weights normalised along the collapsing_coord dimension.
+                The collapsing_coord will be the leading dimension on the
+                output cube.
+        """
+        summed_weights = weights_cube.collapsed(
+            collapsing_coord, iris.analysis.SUM)
+        result = iris.cube.CubeList()
+        # Slice over collapsing_coord so the dimensions match.
+        for weight_slice in (
+                weights_cube.slices_over(collapsing_coord)):
+            # Only divide where the sum of weights are positive. Setting
+            # the out keyword args sets the default value for where
+            # the sum of the weights are less than zero.
+            normalised_data = np.divide(
+                weight_slice.data, summed_weights.data,
+                out=np.zeros_like(weight_slice.data),
+                where=(summed_weights.data > 0))
+            result.append(weight_slice.copy(data=normalised_data))
+        return result.merge_cube()
+
+    def process(self, cube_to_collapse, one_dimensional_weights_cube,
+                collapsing_coord):
+        """
+        Create fuzzy spatial weights based on missing data in the cube we
+        are going to collapse and combine these with 1D weights along the
+        collapsing_coord.
+
+        Args:
+            cube_to_collapse (iris.cube.Cube):
+                The cube that will be collapsed along the collapsing_coord
+                using the spatial weights generated using this plugin. Must
+                be masked where there is invalid data.
+            one_dimensional_weights_cube (iris.cube.Cube):
+                A cube containing a single dimension coordinate with the same
+                name given collapsing_coord. This cube contains 1D weights
+                that will be applied along the collapsing_coord but need
+                adjusting spatially based on missing data.
+            collapsing_coord (string):
+                A string containing the name of the coordinate that the
+                cube_to_collapse will be collapsed along. Also matches the
+                coordinate in one_dimensional_weights_cube.
+
+        Returns:
+            result (iris.cube.Cube):
+                A cube containing normalised spatial weights based on the
+                cube_to_collapsemask and the one_dimensional weights supplied.
+                Contains the same dimensions in the same order as
+                cube_to_collapse.
+        """
+        masked_weights_cube = self.create_initial_weights_from_mask(
+            cube_to_collapse)
+        masked_weights_cube = self.create_fuzzy_spatial_weights(
+            masked_weights_cube)
```

Comment:
I'm going to be flippant and say we're modifying an input (at least conceptually) - so maybe `fuzzify_spatial_weights`?  I feel like the current name implies creating a whole new cube.  (Also I like the word "fuzzify"...)